### PR TITLE
Lawgiver fix firemodes

### PR DESCRIPTION
**Pull request by Ezuo**
The old Lawgiver code didn't work as intended, with different modes using different charge values and having different recoil. This PR fixes that code, so the lawgiver is now somewhat balanced (No more HE spam yay). Current ammo amounts, from full charge, are:

>Stun and Singleshot = 40
>Burst = 13ish, just shoots 3 Singleshot in a row
>AP = 15
>Hotshot and Pellets = 8
>Highex = 5 (Original charge cost placed it at 6.6 recurring, brought it down to 5)

Am open to balance tweaking if required/wanted.

Now uses fire mode stuff instead of strange bullshit.

### CODE REVIEW

**Comment by SonicGotNuked** [comment]
Shouldn't highEx cost higher?

File: code/modules/projectiles/guns/energy/lawgiver.dm [hunk 1]
```diff
@@ -188,5 +213,7 @@
 		message_enabled = 1
 		message_disable = 0
 		play_message()
-	else
-		return
+	
+	var/datum/firemode/new_mode = firemodes[sel_mode]
+	new_mode.apply_to(src)
```
Comment (Arrow768) [inline code/modules/projectiles/guns/energy/lawgiver.dm]: That would get and apply the firemode every time the owner (or anyone else, if emagged) speaks near the lawgiver.

I would put that part into the different if clauses, where the firemodes are changed.
It adds a bit of duplication but avoids calling that proc when not needed.

**Comment by Arrow768** [comment]
I think the highex cost is fine as is.
According to the data provided, you can only get 5 highex shots out of the lawgiver **or** 40 singleshots, if the lawgiver is fully charged.

**Comment by skull132** [comment]
Expedited into #3423 because fuck people using lawgivers to spam ree.

**Pull request closed without merging**